src/cloudimagedirectory/transform/transform.py

```diff
@@ -0,0 +1,67 @@
+from cloudimagedirectory import config
+from cloudimagedirectory.connection import connection
+from cloudimagedirectory.update_images import aws
+
+
+class Pipeline:
+    transformer = []
+    src_conn = None
+
+    def __init__(self, src_conn, transformer_funcs):
+        self.src_conn = src_conn
+        for f in transformer_funcs:
+            self.transformer.append(f(self.src_conn))
+
+    def run(self, data):
+        results = []
+        for t in self.transformer:
+            results.extend(t.run(data))
+        return results
+
+
+class Transformer:
+    src_conn = None
+
+    def __init__(self, src_conn):
+        self.src_conn = src_conn
+
+    def run(self, data):
+        pass
+
+
+class TransformerAWS(Transformer):
+    def __init__(self, src_conn):
+        super().__init__(src_conn)
+
+    def run(self, data):
+        entries = []
+        for d in data:
+            if d.filename.__contains__("aws"):
+                entries.append(d)
+
+        results = []
+        for e in entries:
+            raw = self.src_conn.get_content(e)
+            region = raw.filename.split("/")
+            region = region[len(region) - 1]
+            region = region.split(".")[0]
+            for content in raw.content:
+                if content["OwnerId"] != config.AWS_RHEL_OWNER_ID:
+                    continue
+                r = aws.format_image(content, region)
+                de = connection.DataEntry(
+                    "aws/" + region + "/" + r["name"].replace(" ", "_").lower(), content
```

Comment:
Looking really good! I think there is a small mix up though:
```suggestion
                    "aws/" + region + "/" + r["name"].replace(" ", "_").lower(), r
```

Reply:
Oh thanks! :smiley: Good catch! :rocket: